mls-rs/src/group/commit.rs

```diff
@@ -674,13 +675,7 @@ where
         self.pending_commit = Some(pending_commit);
 
         let ratchet_tree = (!commit_options.ratchet_tree_extension)
-            .then(|| {
-                provisional_state
-                    .public_tree
-                    .export_node_data()
-                    .mls_encode_to_vec()
-            })
-            .transpose()?;
+            .then(|| ExportedTree::new(provisional_state.public_tree.export_node_data()));
 
         if let Some(signer) = new_signer {
```

Comment:
This clones the tree while it could move it out, since `provisional_state` is later deleted.